spec/bundler/shared_helpers_spec.rb

```diff
@@ -170,6 +170,105 @@
       end
     end
   end
+  describe "#in_bundle?" do
+    it "calls to find_gemfile" do
+      expect(subject).to receive(:find_gemfile)
+      subject.in_bundle?
+    end
+  end
+  describe "#chdir" do
+    before do
+      Dir.mkdir "chdir_test_dir"
+      ext_lock_double = double(:ext_lock)
+      allow(Bundler.rubygems).to receive(:ext_lock).and_return(ext_lock_double)
+      allow(ext_lock_double).to receive(:synchronize) do |&block|
+        block.call
+      end
+    end
+    it "executes the passed block while in the specified directory" do
+      op_block = proc { Dir.mkdir "nested_dir" }
+      subject.chdir("chdir_test_dir", &op_block)
+      expect(Dir.exist?("chdir_test_dir/nested_dir")).to eq(true)
+    end
+  end
+  describe "#pwd" do
+    before do
+      ext_lock_double = double(:ext_lock)
+      allow(Bundler.rubygems).to receive(:ext_lock).and_return(ext_lock_double)
+      allow(ext_lock_double).to receive(:synchronize) do |&block|
+        block.call
+      end
+    end
+    it "returns the current absolute path" do
+      expect(subject.pwd).to eq(bundled_app)
+    end
+  end
+  describe "#filesystem_access" do
+    context "system has proper permission access" do
+      it "performs the operation in the passed block" do
+        file_op_block = proc {|path| FileUtils.mkdir_p(path) }
+        subject.filesystem_access("./test_dir", &file_op_block)
+        expect(Dir.exist?("./test_dir")).to eq(true)
```

Comment:
Not sure what you mean, could you clarify? Like: `expect(File).to exist("chdir_test_dir/nested_dir")` ? Or something with the `Pathname` class ?